fe/cycles.py

```diff
@@ -0,0 +1,155 @@
+from jax import config
+
+config.update("jax_enable_x64", True)
+
+from jax import grad, numpy as np
+import numpy as onp
+
+import cvxpy as cp
+from cvxpylayers.jax import CvxpyLayer
+
+from fe.utils import validate_map
+
+from typing import Optional
+
+
+def construct_mle_layer(n_nodes: int,
+                        rbfe_inds: np.array, rbfe_sigmas: Optional[np.array] = None,
+                        abfe_inds: Optional[np.array] = None, abfe_sigmas: Optional[np.array] = None) -> callable:
+    """Construct a differentiable function predict_fs(simulated_rbfes, simulated_abfes) -> fs
```

Comment:
Does `fs` here stand for free energies?

Reply:
The naming in this function could stand to be improved -- probably best to get this right early!

Would something like `rbfe_inds` -> `relative_inds`, `abfe_inds` -> `absolute_inds`, `fs` -> `free_energies` be an improvement? I'm also considering to replace the paired arguments `(rbfe_inds, rbfe_sigmas)` with an "RBFEInfo" object, `(abfe_inds, abfe_sigmas)` with an "ABFEInfo" object, or similar, which may improve readability.

Reply:
The only one that was opaque to me was `fs`, the others seem in our wheel house already

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/399/commits/d604200579747051844e9f3ba9fa4bb4ed9c6c90